torch/csrc/lazy/ts_backend/ts_eager_fallback.cpp

```diff
@@ -68,6 +69,35 @@ std::vector<at::Tensor> to_eager(
   return eager_tensors;
 }
 
+std::vector<c10::optional<at::Tensor>> to_eager(
+    const std::vector<c10::optional<at::Tensor>>& tensors,
+    c10::DeviceType device_type) {
+  // We can't just call _to_eager() on the entire list of Tensors because it
+  // will break on undefined tensors. Separate out undefined tensors first.
```

Comment:
I'm confused -- std::vector<c10::optional<at::Tensor>> probably came from an IValue that is `List[Optional[Tensor]]`, but it can contain undefined tensors?

Reply:
it's just copy pasted from the impl above lol

Reply:
I think it should be possible to assume that inside optional they are never undefined, and things should work